### DISCUSSION TITLE
Table block, query with pagination and/or column look-up

### DISCUSSION THREAD

**Discussion by yuppox** [Budibase/budibase#8959] [Help]
I'm trying Budibase for the first time. Is it correct that queries used as Data cannot paginate? If so, why?

I'm using MySQL as the data source. I would like to use a query but if there's a way to configure look-up columns, I can use the table. Some of the values are stored by a reference and need another table to make it user-friendly. Is that possible?

EDIT:

I see that Budibase has "relationships" but I am struggling with that option. I can link the primary key of a table to a reference in another table. However, then all I see is the primary key. For example:

```
table: Status (id, friendly)
  1  Down
  2  Up

table: Server (id, name, status)
  1  WebServer   2
  2  TestServer  1
```

I would create a relationship between `Status` and `Server` where `Status: id` is linked to `Server: status`. How do I tell it to show the `friendly` column instead of `id`?

**Comment by melohagan** [comment]
Hey @yuppox 

That's right, queries used in a Data provider cannot be paginated, however you can build your own pagination: https://github.com/Budibase/budibase/discussions/7313#discussioncomment-3430353

The 'friendly' column can be shown by setting the table display column: https://docs.budibase.com/docs/budibasedb#selecting-the-display-column
